tests/foreman/api/test_discoveredhost.py

```diff
@@ -68,561 +76,547 @@ def _create_discovered_host(name=None, ipaddress=None, macaddress=None):
-            )
-            if discovered_host:
-                break
-            time.sleep(10)
-        else:
-            raise HostNotDiscoveredException(
-                "The host {} is not discovered within 300 seconds".format(hostname)
-            )
-        return discovered_host[0]
-
-    @classmethod
-    @skip_if_not_set('vlan_networking')
-    def setUpClass(cls):
-        """Steps to Configure foreman discovery
-
-        1. Build PXE default template
-        2. Create Organization/Location
-        3. Update Global parameters to set default org and location for
-           discovered hosts.
-        4. Enable auto_provision flag to perform discovery via discovery
-           rules.
-        """
-        super(DiscoveryTestCase, cls).setUpClass()
-
-        # Build PXE default template to get default PXE file
-        entities.ProvisioningTemplate().build_pxe_default()
-        # let's just modify the timeouts to speed things up
-        ssh.command(
-            "sed -ie 's/TIMEOUT [[:digit:]]\\+/TIMEOUT 1/g' "
-            "/var/lib/tftpboot/pxelinux.cfg/default"
+    Introduced a delay of 300secs by polling every 10 secs to get expected
+    host
+    """
+    # assert that server receives DHCP discover from hosts PXELinux
+    for pattern in [
+        (f"DHCPDISCOVER from {host.mac}", "DHCPDISCOVER",),
+        (f"DHCPACK on [0-9.]+ to {host.mac}", "DHCPACK"),
+    ]:
+        try:
+            dhcp_pxe = _wait_for_log(channel, pattern[0], timeout=30)
+        except TimedOutError:
+            # raise assertion error
+            raise AssertionError(f'Timed out waiting for {pattern[1]} from VM')
+
+    groups = re.search('DHCPACK on (\\d.+) to', dhcp_pxe.out)
+    assert len(groups.groups()) == 1, 'Unable to parse bootloader ip address'
+    pxe_ip = groups.groups()[0]
+
+    # assert that server retrieves pxelinux config over TFTP
+    for pattern in [
+        (f'Client {pxe_ip} finished pxelinux.0', 'pxelinux.0'),
+        (f'Client {pxe_ip} finished pxelinux.cfg/default', 'pxelinux.cfg/default'),
+        (f'Client {pxe_ip} finished boot/fdi-image/vmlinuz0', 'fdi-image/vmlinuz0'),
+        (f'Client {pxe_ip} finished boot/fdi-image/initrd0.img', 'fdi-image/initrd0.img'),
+    ]:
+        try:
+            _wait_for_log(channel, pattern[0], timeout=120)
+        except TimedOutError:
+            # raise assertion error
+            raise AssertionError(f'Timed out waiting for VM (tftp) to fetch {pattern[1]}')
+
+    # assert that server receives DHCP discover from FDI
+    for pattern in [
+        (f'DHCPDISCOVER from {host.mac}', 'DHCPDISCOVER',),
+        (f'DHCPACK on [0-9.]+ to {host.mac}', 'DHCPACK'),
+    ]:
+        try:
+            dhcp_fdi = _wait_for_log(channel, pattern[0], timeout=30)
+        except TimedOutError:
+            # raise assertion error
+            raise AssertionError(f'Timed out waiting for {pattern[1]} from VM')
+    groups = re.search('DHCPACK on (\\d.+) to', dhcp_fdi.out)
```

Comment:
I'd recommend using `r''` string prefix here to avoid warnings on escape characters.

Reply:
Applies to all regex expressions in this PR.